### PR TITLE
Restrict ModManagement commands to moderation channels

### PR DESCRIPTION
**Pull request by atmishra**
Closes:  #535
This PR add an extra check `in_channel_check(ctx, *constants.MODERATION_CHANNELS)` to `cog_check`. This conditon will pass only if the channel belongs to the list of moderation channels.

### CODE REVIEW

File: bot/cogs/moderation/management.py [hunk 1]
```diff
@@ -257,7 +257,11 @@ def infraction_to_string(self, infraction: utils.Infraction) -> str:
     # This cannot be static (must have a __func__ attribute).
     def cog_check(self, ctx: Context) -> bool:
         """Only allow moderators to invoke the commands in this cog."""
```
Comment (lemonsaurus) [inline bot/cogs/moderation/management.py]: This docstring needs to be updated to reflect the new check.

Comment (atmishra) [reply bot/cogs/moderation/management.py]: Thanks! will do the changes.

**Comment by atmishra** [comment]
I wasn't able to test it and need some help with that.

**Comment by MarkKoz** [comment]
@atmishra If you need help don't hesitate to leave your question in the #meta channel on Discord.

File: bot/cogs/moderation/management.py (hunk 1 above)
Comment (atmishra) [reply bot/cogs/moderation/management.py]: Updated the doc-string. I am not familiar with terminology yet so, not sure if doc-string makes much sense.

**Comment by atmishra** [comment]
@MarkKoz Thanks! Will post my queries there.

**Comment by atmishra** [comment]
I tested this manually by calling the `!infractions` command. I found an issue, `in_channel_check` just takes a single channel and I am trying to pass a list of channels to it in my PR. Should I modify the `in_channel_check` implementation to support list of channels or just loop over different channels and call this method for each of them?

**Comment by MarkKoz** [comment]
Preferably modify `in_channel_check`. There are also unit tests for this that would need to be adjusted/added. Also there is no need for a loop because the `in` operator can be used.

**Comment by atmishra** [comment]
I have done the changes required. I also did manual testing and this works as expected.

**Pull request merged**